### PR TITLE
BUG: Fix Series.is_unique with single occurrence of NaN

### LINKED ISSUE #25180: BUG: Series.is_unique does not consider single occurrences of NaN unique
#### Code Sample, a copy-pastable example if possible
On `master`:
```python
In [1]: import numpy as np; import pandas as pd; pd.__version__
Out[1]: '0.25.0.dev0+67.gf04bb2fc9'

In [2]: pd.Series(['foo', 'bar', np.nan]).is_unique
Out[2]: False

In [3]: pd.Series([np.nan]).is_unique
Out[3]: False
```

Note that indexes consider the equivalent cases to be unique:
```python
In [4]: pd.Index(['foo', 'bar', np.nan]).is_unique
Out[4]: True

In [5]: pd.Float64Index([np.nan]).is_unique
Out[5]: True
```
#### Problem description
`Series.is_unique` does not consider single occurrences of `NaN` to be unique.

#### Expected Output
I'd expect `[2]` and `[3]` to be `True`.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: f04bb2fc9c3d0d23e92bbdf09fe6b94e59589cbe
python: 3.6.8.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 78 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.25.0.dev0+67.gf04bb2fc9
pytest: 4.0.2
pip: 9.0.1
setuptools: 40.6.3
Cython: 0.28.2
numpy: 1.14.6
scipy: 1.0.0
pyarrow: 0.6.0
xarray: 0.9.6
IPython: 7.2.0
sphinx: 1.8.2
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.4
feather: 0.4.0
matplotlib: 2.0.2
openpyxl: 2.4.8
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 0.9.8
lxml.etree: 3.8.0
bs4: None
html5lib: 0.999
sqlalchemy: 1.1.13
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
fastparquet: 0.1.5
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by jschendel**
- [X] closes #25180
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

### CODE REVIEW

File: pandas/tests/series/test_duplicates.py [hunk 1]
```diff
@@ -59,12 +59,19 @@ def test_unique_data_ownership():
     Series(Series(["a", "c", "b"]).unique()).sort_values()
 
 
-def test_is_unique():
-    # GH11946
-    s = Series(np.random.randint(0, 10, size=1000))
-    assert s.is_unique is False
-    s = Series(np.arange(1000))
-    assert s.is_unique is True
+@pytest.mark.parametrize('data, expected', [
```
Comment (jreback) [inline pandas/tests/series/test_duplicates.py]: do we have sufficient tests on the Index side for this? (e.g. your parameterization is good, maybe add for the index tests)/

**Comment by jreback** [comment]
lgtm. if we have sufficient tests, merge on green, otherwise if you can add.

**Comment by jschendel** [comment]
@jreback : Added index tests.  Didn't see many `is_unique` tests for index; some indexes tested it but other didn't appear to.  Added a test to the base class to test `is_unique` in general, as well as the `NaN` specific issue.  Removed some tests that were made redundant by the common test.

**Comment by jreback** [comment]
thanks!

**Pull request merged**